controllers/numaresourcesoperator_controller_test.go

```diff
@@ -468,7 +470,66 @@ var _ = Describe("Test NUMAResourcesOperator Reconcile", func() {
 						}
 						Expect(reconciler.Client.Get(context.TODO(), mcp2DSKey, ds)).ToNot(HaveOccurred())
 					})
-
+					When(" daemonsets are ready", func() {
+						BeforeEach(func() {
+							key := client.ObjectKeyFromObject(nro)
+
+							reconciler.RTEManifests.DaemonSet.Status.DesiredNumberScheduled = int32(len(nro.Spec.NodeGroups))
+							reconciler.RTEManifests.DaemonSet.Status.NumberReady = reconciler.RTEManifests.DaemonSet.Status.DesiredNumberScheduled
```

Comment:
this works (but is not great) but pollutes the global state - we should at least restore the initial values

Reply:
yeah I am not a fan of this neither but I did not found another way to do it. 
if there is a better way please let me know and I will change it with pleasure.

About restore initial values, you're totally right ... on it.